app/utils/openlayers.SearchControl.js

```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
 
 export default class SearchControl extends openlayers.control.Control {
   constructor(className) {
+
     const element = document.createElement('div')
     super({element})
     element.className = `${className || ''} ol-unselectable ol-control`
     element.title = 'Jump to a specific location or coordinate'
+    element.id = 'searchCoordBtn'
     element.innerHTML = '<button><i class="fa fa-search"/></button>'
-    element.addEventListener('click', () => this._clicked())
+    element.addEventListener('click', () => this._searchClicked())
+
   }
 
-  _clicked() {
-    console.debug('things')
+
+  getDialog(){
+    if(!this._dialog){
+      this._dialog = document.createElement('form')
+      this._dialog.className = 'coordinate-dialog'
+      this._dialog.style.display = 'block'
+      this._dialog.style.position = 'absolute'
+      this._dialog.style.top = '300px'
+      this._dialog.style.left = '50%'
+      this._dialog.style.transform = 'translateX(-50%)'
+      this._dialog.innerHTML = '<input style="width: 400px; height: 26px;" name="coordinate">&nbsp<button style="vertical-align: bottom"; type="submit"><i class="fa fa-search fa-2x"/></button>'
+      this._dialog.addEventListener('submit', (event) => this._formSubmitted(event))
+      this.getMap().getTarget().appendChild(this._dialog)
+    }
+    return this._dialog
+  }
+
+  _formSubmitted(event){
+
+    console.log(this._dialog.querySelector("input").value)
+    event.preventDefault()
+
+    var [x,y] = this._convertGeospatialCoordinateFormat(this._dialog.querySelector("input").value)
+    var lat = parseFloat(y)
+    var long = parseFloat(x)
+    var map = this.getMap()
+
+    var view = map.getView()
+    view.setCenter(openlayers.proj.transform([long, lat], 'EPSG:4326', 'EPSG:3857'))
+    view.setZoom(8)
+    this._dialog.style.display = 'none'
+    this._dialog.querySelector("input").value = ""
+  }
+
+  _searchClicked() {
+    this.getDialog().style.display = 'block'
+  }
+
+  _convertGeospatialCoordinateFormat(inputString) {
+
+    var ddPattern = /(\-?\d{1,2}[.]?\d*)[\s+|,?]\s*(\-?\d{1,3}[.]?\d*)/
+    var dmsPattern = /(\d{2})[^\d]*(\d{2})[^\d]*(\d{2}[.]?\d*)([n|N|s|S])[^\w]*(\d{3})[^\d]*(\d{2})[^d]*(\d{2}[.]?\d*)([e|E|w|W])/
+    var mgrsPattern = /(\d{1,2})([a-zA-Z])[^\w]*([a-zA-Z])([a-zA-Z])[^\w]*(\d{5})[^\w]*(\d{5})/
+
+    if (inputString.match(dmsPattern)) {
+      var latitude = this.dmsToDd(RegExp.$1, RegExp.$2, RegExp.$3, RegExp.$4)
```

Comment:
I don't think we want to use `RegEx.$1`.

I have honestly never seen this before and MDN also recommends against their use:
![screen shot 2016-06-09 at 12 44 26 pm](https://cloud.githubusercontent.com/assets/1542531/15938166/058d8032-2e40-11e6-92da-e9e1bdffcac2.png)